src/hooks/useHandleExceedMaxCommentLength.ts

```diff
@@ -0,0 +1,27 @@
+import _ from 'lodash';
+import {useCallback, useMemo, useState} from 'react';
+import * as ReportUtils from '@libs/ReportUtils';
+import CONST from '@src/CONST';
+
+const useHandleExceedMaxCommentLength = () => {
+    const [hasExceededMaxCommentLength, setHasExceededMaxCommentLength] = useState(false);
+
+    const handleValueChange = useCallback(
+        (value: string) => {
+            if (ReportUtils.getCommentLength(value) <= CONST.MAX_COMMENT_LENGTH) {
+                if (hasExceededMaxCommentLength) {
+                    setHasExceededMaxCommentLength(false);
+                }
+                return;
+            }
+            setHasExceededMaxCommentLength(true);
+        },
+        [hasExceededMaxCommentLength],
+    );
+
+    const handleValueChangeDebounce = useMemo(() => _.debounce(handleValueChange, 1500), [handleValueChange]);
+
+    return {hasExceededMaxCommentLength, handleValueChangeDebounce};
```

Comment:
Could you please rename `handleValueChangeDebounce` to `validateCommentMaxLength`? There are a few reasons for this:

- Functions should be named for what they do, not what they handle
- It isn't necessary for anything to know that it's debounced (it's an implementation detail)
- It is a bit ambiguous without mentioning the word "comment"

Reply:
@tgolen I updated